src/storage/src/render.rs

```diff
@@ -249,54 +252,83 @@ pub fn build_ingestion_dataflow<A: Allocate>(
             let (feedback_handle, feedback) = mz_scope.feedback(Default::default());
 
             let connection = description.desc.connection.clone();
+
+            let busy_signal = if dyncfgs::SUSPENDABLE_SOURCES
+                .get(storage_state.storage_configuration.config_set())
+            {
+                Arc::new(Semaphore::new(1))
```

Comment:
this may be something addressed in the follow up PR, but wouldn't `1` be overly restrictive here? We would essentially block to polling one future at a time across all threads?

Reply:
Oh _duh_! Sorry at first glance I thought this was at a global scope. makes sense!